File: app/src/main/java/com/mxt/anitrend/view/fragment/list/AiringListFragment.java
package com.mxt.anitrend.view.fragment.list;

import android.os.Bundle;
import androidx.annotation.Nullable;

import com.annimon.stream.Optional;
import com.annimon.stream.Stream;
import com.mxt.anitrend.adapter.recycler.index.MediaListAdapter;
import com.mxt.anitrend.model.entity.anilist.MediaList;
import com.mxt.anitrend.model.entity.anilist.MediaListCollection;
import com.mxt.anitrend.model.entity.base.UserBase;
import com.mxt.anitrend.model.entity.container.body.PageContainer;
import com.mxt.anitrend.util.CompatUtil;
import com.mxt.anitrend.util.GraphUtil;
import com.mxt.anitrend.util.KeyUtil;
import com.mxt.anitrend.util.MediaListUtil;

import java.util.Collections;
import java.util.List;

/**
 * Created by max on 2017/11/03.
 */

public class AiringListFragment extends MediaListFragment {

    public static AiringListFragment newInstance() {
        return new AiringListFragment();
    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        UserBase userBase = getPresenter().getDatabase().getCurrentUser();
        userId = userBase.getId(); userName = userBase.getName(); mediaType = KeyUtil.Companion.getANIME();
        ((MediaListAdapter) getMAdapter()).setCurrentUser(userName);
        queryContainer = GraphUtil.INSTANCE.getDefaultQuery(false)
                .putVariable(KeyUtil.Companion.getArg_statusIn(), KeyUtil.Companion.getCURRENT());
    }

    /**
     * Is automatically called in the @onStart Method if overridden in list implementation
     */
    @Override
    protected void updateUI() {
        injectAdapter();
    }

    @Override
    public void onChanged(@Nullable PageContainer<MediaListCollection> content) {
        if(content != null) {
            if(content.hasPageInfo())
                getPresenter().setPageInfo(content.getPageInfo());
            if(!content.isEmpty()) {
                Optional<MediaListCollection> mediaOptional = Stream.of(content.getPageData()).findFirst();
                if(mediaOptional.isPresent()) {
                    MediaListCollection mediaListCollection = mediaOptional.get();

                    List<MediaList> mediaList = Stream.of(mediaListCollection.getEntries())
                            .filter(media -> CompatUtil.INSTANCE.equals(media.getMedia().getStatus(), KeyUtil.Companion.getRELEASING()))
                            .toList();

                    if(MediaListUtil.isTitleSort(getPresenter().getSettings().getMediaListSort()))
                        sortMediaListByTitle(mediaList);
                    else
                        onPostProcessed(mediaList);
                    mediaListCollectionBase = mediaListCollection;
                } else
                    onPostProcessed(Collections.emptyList());
            }
            else
                onPostProcessed(Collections.emptyList());
        } else
            onPostProcessed(Collections.emptyList());
        if(getMAdapter().getItemCount() < 1)
            onPostProcessed(null);
    }
}

File: app/src/main/java/com/mxt/anitrend/view/fragment/list/AiringListFragment.kt
package com.mxt.anitrend.view.fragment.list

import android.os.Bundle

import com.annimon.stream.Optional
import com.annimon.stream.Stream
import com.mxt.anitrend.adapter.recycler.index.MediaListAdapter
import com.mxt.anitrend.model.entity.anilist.MediaList
import com.mxt.anitrend.model.entity.anilist.MediaListCollection
import com.mxt.anitrend.model.entity.base.UserBase
import com.mxt.anitrend.model.entity.container.body.PageContainer
import com.mxt.anitrend.util.CompatUtil
import com.mxt.anitrend.util.GraphUtil
import com.mxt.anitrend.util.KeyUtil
import com.mxt.anitrend.util.MediaListUtil

import java.util.Collections

/**
 * Created by max on 2017/11/03.
 */

class AiringListFragment : MediaListFragment() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val userBase = presenter.database.currentUser
        userId = userBase!!.id
        userName = userBase.name
        mediaType = KeyUtil.getANIME()
        (mAdapter as MediaListAdapter).setCurrentUser(userName!!)
        queryContainer = GraphUtil.getDefaultQuery(false)
            .putVariable(KeyUtil.getArg_statusIn(), KeyUtil.getCURRENT())
    }

    /**
     * Is automatically called in the @onStart Method if overridden in list implementation
     */
    override fun updateUI() {
        injectAdapter()
    }

    override fun onChanged(content: PageContainer<MediaListCollection>?) {
        if (content != null) {
            if (content.hasPageInfo())
                presenter.pageInfo = content.pageInfo
            if (!content.isEmpty) {
                val mediaOptional = Stream.of(content.pageData).findFirst()
                if (mediaOptional.isPresent) {
                    val mediaListCollection = mediaOptional.get()

                    val mediaList = Stream.of(mediaListCollection.entries)
                        .filter { media -> CompatUtil.equals(media.media.status, KeyUtil.getRELEASING()) }
                        .toList()

                    if (MediaListUtil.isTitleSort(presenter.settings.mediaListSort))
                        sortMediaListByTitle(mediaList)
                    else
                        onPostProcessed(mediaList)
                    mediaListCollectionBase = mediaListCollection
                } else
                    onPostProcessed(emptyList())
            } else
                onPostProcessed(emptyList())
        } else
            onPostProcessed(emptyList())
        if (mAdapter!!.itemCount < 1)
            onPostProcessed(null)
    }

    companion object {

        fun newInstance(): AiringListFragment {
            return AiringListFragment()
        }
    }
}
